### ISSUE TITLE
FacebookClient isn't derivied from an interface

### ISSUE THREAD

**Issue by junderhill** [facebook-csharp-sdk/facebook-csharp-sdk#354]
Whilst trying to mock the FacebookClient in my application I found that it doesn't have an interface which I can generate a mock against. 

Is this by design or is this something that you'd be happy for me to create a pull request for?

Thanks

**Comment by prabirshrestha** [comment]
methods are virtual so you should be able to mock it.

You can see some samples here https://github.com/prabirshrestha/Facebook.Moq

**Comment by junderhill** [comment]
@prabirshrestha That's true although doesn't the lack of interface force developers to deviate from the 5th SOLID principle by them having to depend on concrete implementations? This was what sprung to mind when I was looking for an interface for FacebookClient to code against.
